### PR TITLE
Update the Resctrictions section for Vector Search

### PR DESCRIPTION
**Pull request by rpaik**
Add info about conversion to single-precision floats until double precision is supported.

(per [this discussion](https://discord.com/channels/1083300679386406923/1098161518182400021/threads/1273819443847495832) during the TiDB Future App Hackathon). 

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by breezewish — APPROVED** [review]

**Review by hfxsd — APPROVED** [review]

File: tidb-cloud/vector-search-data-types.md [hunk 1]
```diff
@@ -238,7 +238,7 @@ It is currently not possible to cast between Vector and other data types (like `
 
 - The maximum supported Vector dimension is 16000.
 - You cannot store `NaN`, `Infinity`, or `-Infinity` values in the vector data type.
-- Currently Vector data types cannot store double-precision floating numbers. This will be supported in future release.
+- Currently Vector data types cannot store double-precision floating point numbers. This will be supported in a future release. In the meantime, if you import double-precision floating point numbers for Vector data types, they will be converted to single-precision numbers.
```
Comment (lilin90) [inline tidb-cloud/vector-search-data-types.md:241]:
```suggestion
- Currently, Vector data types cannot store double-precision floating point numbers. This is planned to be supported in a future release. In the meantime, if you import double-precision floating point numbers for Vector data types, they are converted to single-precision numbers.
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**